File: files/CH-094.py
import pandas as pd

path = "CH-094 Two Column Text.xlsx"
input = pd.read_excel(path, skiprows=1, usecols = "B:C")
test  = pd.read_excel(path, skiprows=1, usecols = "H:J", nrows = 7)
test.columns = test.columns.str.replace(".1", "")

result = input.copy()
result["RowNumber"] = result.groupby("Group").cumcount() + 1
result["col"] = result["RowNumber"].apply(lambda x: 2 if x % 2 == 0 else 1)
result["row"] = result["RowNumber"].apply(lambda x: (x + 1) // 2)

result = result.pivot(index = ["Group", "row"], columns = "col", values = "Text")
result.columns = [f"Column {col}" for col in result.columns]
result = result.reset_index()
result = result.drop(columns = "row")

print(result.equals(test)) # True

File: files/000-099/094/CH-094.R
library(tidyverse)
library(readxl)

path = "files/CH-094 Two Column Text.xlsx"
input = read_excel(path, range = "B2:C13")
test  = read_excel(path, range = "H2:J9")

result = input %>%
  mutate(column = ifelse(row_number() %% 2 == 0, 2, 1),
         row = ceiling(row_number() / 2), 
         .by = Group) %>%
  pivot_wider(names_from = column, values_from = Text, names_glue = "Column {column}") %>%
  select(-row)

all.equal(result, test) 
#> [1] TRUE 